### ISSUE TITLE
Question about the split of train/valid/test data

### ISSUE THREAD

**Issue by sunxiaoyao-git** [LarsBentsen/FFTransformer#10]
Hi, 
   I read the paper and know that "The first 60% of the data was used for training, whereas the following 20% and the remainder were used for validation and testing", at the same time you didn`t interpolate all time-series. So, my question is that whether you considered the missing value when you do the dataset splitting.
  I look your code, which don`t explain the splitting process and just give three files for train/valid/test.
 Looking forward to your reply and thank you  again!

Below, I try to split the data through skipping missing values 
        # train_end = int(df_raw.dropna().count().min() * 0.7)
        # test_end = int(df_raw.dropna().count().min() * 0.1)
        # vali_end = df_raw.dropna().count().min() - test_end
        # num_train = 0
        # num_valid = 0
        # count_non_null = 0
        # for index, row in df_raw.iterrows():
        #     if row.notna().all():
        #         count_non_null += 1
        #         if count_non_null == train_end:
        #             num_train = index + 1
        #             break
        # count_non_null = 0
        # for index, row in df_raw.iterrows():
        #     if row.notna().all():
        #         count_non_null += 1
        #         if count_non_null == vali_end:
        #             num_valid = index + 1
        #             break
        # border1s = [0, num_train - self.seq_len, num_valid - self.seq_len]
        # border2s = [num_train, num_valid, len(df_raw)]
        # border1 = border1s[self.set_type]
        # border2 = border2s[self.set_type]

**Comment by LarsBentsen** [comment]
Hi! 
When creating the three datasets, I include samples where a single value is missing and set this value to be in between the last and next value, i.e. if v_t is missing then I set v_t = (v_{t+1} + v_{t-1})/2. I then split the data (without considering the amount of NaN values in the respective sets). For dealing with NaN values, I only consider batches without NaN values as can be seen in the `data_loader.py` file. 
I hope this answers your question and if so you can close the issue. Otherwise, please let me know if there's anything else I can help you with!

**Comment by sunxiaoyao-git** [comment]
Thanks a lot!
For me, the missing values in my data are not evenly distributed, resulting in the amount of train/valid/test not being 7:2:1 as I set.  For example, 22% missing values of train set, 10% missing values of valid set, and 2%missing of test set. I wonder if this would influence the accuracy of model?

**Comment by LarsBentsen** [comment]
It's difficult to say precisely how this will influence accuracy. 
However, with your amounts of missing data, you could try a different split (than the 60:20:20 that I used), for example, 70:15:15, 7:2:1 (as you suggested) or 8:1:1. 
Nevertheless, I would be careful to not have the validation and test sets too small, as you might then only include particular seasons in the test and validation sets, which might have distinct characteristics. In other words, if the test and validation sets are too small with the time-splitting approach that we used in this paper, you might not be able to evaluate the models properly.

**Comment by sunxiaoyao-git** [comment]
I get it. Thank you~~
